python/lsst/pipe/tasks/dcrAssembleCoadd.py

```diff
@@ -0,0 +1,980 @@
+            coaddUtils.setCoaddEdgeBits(model.mask, model.variance)
+            coaddExposure.setMaskedImage(model)
+            dcrCoadds.append(coaddExposure)
+        return dcrCoadds
+
+    def convolveDcrModelPlane(self, maskedImage, dcr, bbox=None, useInverse=False):
+        """Shift a masked image.
+
+        Parameters
+        ----------
+        maskedImage : lsst.afw.image.maskedImageF
+            The input masked image to shift.
+        dcr : lsst.afw.geom.Extent2I
+            Shift calculated with `dcrShiftCalculate`.
+        bbox : lsst.afw.geom.box.Box2I, optional
+            Sub-region of the masked image to shift.
+            Shifts the entire image if None.
+        useFFT : bool, optional
+            Perform the convolution with an FFT?
+        useInverse : bool, optional
+            Use the reverse of `dcr` for the shift.
+
+        Returns
+        -------
+        lsst.afw.image.maskedImageF
+            A masked image, with the pixels within the bounding box shifted.
+
+        Raises
+        ------
+        NotImplementedError
+            The Fourier transform approach has not yet been written.
+        """
+        if self.config.useFFT:
+            raise NotImplementedError("The Fourier transform approach has not yet been written.")
+        else:
+            padValue = afwImage.pixel.SinglePixelF(0., maskedImage.mask.getPlaneBitMask("NO_DATA"), 0)
+            if bbox is None:
+                bbox = maskedImage.getBBox()
+            shiftedImage = afwImage.MaskedImageF(bbox)
+            if useInverse:
+                transform = makeTransform(AffineTransform(-dcr))
+            else:
+                transform = makeTransform(AffineTransform(dcr))
+            afwMath.warpImage(shiftedImage, maskedImage[bbox, afwImage.PARENT],
+                              transform, self.warpCtrl, padValue=padValue)
+        return shiftedImage
+
+    @staticmethod
+    def conditionDcrModel(oldDcrModels, newDcrModels, bbox, gain=1.):
+        """Average two iterations' solutions to reduce oscillations.
+
+        Parameters
+        ----------
+        oldDcrModels : list of lsst.afw.image.maskedImageF
+            The models for each subfilter from the previous iteration.
+        newDcrModels : list of lsst.afw.image.maskedImageF
+            The models for each subfilter from the current iteration.
+        bbox : lsst.afw.geom.box.Box2I
+            Sub-region of the coadd
+        gain : `float`, optional
+            Additional weight to apply to the model from the current iteration.
+        """
+        for oldModel, newModel in zip(oldDcrModels, newDcrModels):
+            # The DcrModels are MaskedImages, which only support in-place operations.
+            newModel *= gain
+            newModel += oldModel[bbox, afwImage.PARENT]
+            newModel.image.array[:] /= 1. + gain
+            newModel.variance.array[:] /= 1. + gain
+
+    def dcrShiftCalculate(self, visitInfo, wcs):
```

Comment:
Naming rules say put the verb first: `calculateDcrShift`.

And isn't DCR tautologically a shift? Then `calculateDcr`.